db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/subquery/SubqueryNodeFactory.java

```diff
@@ -105,10 +105,10 @@ public FromSubquery buildFromSubqueryNode(SelectNode outerSelectNode,
      */
     public HalfOuterJoinNode buildOuterJoinNode(FromList outerFromList,
                                                 FromSubquery fromSubquery,
-                                                ValueNode joinClause) throws StandardException {
+                                                ValueNode joinClause,
+                                                FromTable outerFromTable) throws StandardException {
         /* Currently we only attempt not-exist flattening if the outer table has one table. */
-        assert outerFromList.size() == 1 : "expected only one FromList element at this point";
-        QueryTreeNode outerTable = outerFromList.getNodes().get(0);
+        FromTable outerTable = outerFromTable == null ? (FromTable)outerFromList.getNodes().get(0) : outerFromTable;
```

Comment:
Is the comment still valid after the assertion is removed?

Reply:
Updated the comment to:

>         /* If outerFromTable is null, that means there is only one outer table.
>          * If outerFromTable is not null, it specifies the outer table we are flattening.
>          */

Reply:
Is it possible to have a case where `outerFromList.size() > 1`?

Reply:
Yes.  This was a severe limitation to flattening before, where we only supported outerFromList.size() == 1.  With DB-11925, this restriction is lifted.